dex/elrond_dex_router/src/pair_manager.rs

```diff
@@ -0,0 +1,198 @@
+            .execute_on_dest_context();
+
+        Ok(())
+    }
+
+    #[endpoint(setFeeOff)]
+    fn set_fee_off(
+        &self,
+        pair_address: Address,
+        fee_to_address: Address,
+        fee_token: TokenIdentifier,
+    ) -> SCResult<()> {
+        self.require_owner()?;
+        require!(self.is_active(), "Not active");
+        self.check_is_pair_sc(&pair_address)?;
+
+        self.pair_contract_proxy(pair_address)
+            .setFeeOn(false, fee_to_address, fee_token)
+            .execute_on_dest_context();
+
+        Ok(())
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn acceptPay(&self) {}
+
+    #[payable("*")]
+    #[endpoint]
+    fn multiPairSwap(
+        &self,
+        #[payment_token] token_id: TokenIdentifier,
+        #[payment_amount] amount: Self::BigUint,
+        #[payment_nonce] nonce: Nonce,
+        swap_operations: MultiArgVec<SwapOperationType<Self::BigUint>>,
+        #[var_args] opt_accept_funds_func: OptionalArg<BoxedBytes>,
+    ) -> SCResult<()> {
+        require!(nonce == 0, "Invalid nonce. Should be zero");
+        require!(amount > 0, "Invalid amount. Should not be zero");
+        require!(
+            !swap_operations.is_empty(),
+            "Invalid swap operations chain. Should not be empty"
+        );
+
+        let caller = self.blockchain().get_caller();
+        let mut residuum_vec = Vec::new();
+        let mut last_received_token_id = token_id;
+        let mut last_received_amount = amount;
+
+        for entry in swap_operations.into_vec() {
+            let (pair_address, function, token_wanted, amount_wanted) = entry.into_tuple();
+            self.check_is_pair_sc(&pair_address)?;
+
+            if function == BoxedBytes::from(SWAP_TOKENS_FIXED_INPUT_FUNC_NAME) {
+                let token_amount_out = self.actual_swap_fixed_input(
+                    pair_address,
+                    last_received_token_id,
+                    last_received_amount,
+                    token_wanted,
+                    amount_wanted,
+                );
+                last_received_token_id = token_amount_out.token_id;
+                last_received_amount = token_amount_out.amount;
+            } else if function == BoxedBytes::from(SWAP_TOKENS_FIXED_OUTPUT_FUNC_NAME) {
+                let (token_amount_out, residuum) = self.actual_swap_fixed_output(
+                    pair_address,
+                    last_received_token_id,
+                    last_received_amount,
+                    token_wanted,
+                    amount_wanted,
+                );
+                last_received_token_id = token_amount_out.token_id;
+                last_received_amount = token_amount_out.amount;
+                residuum_vec.push(residuum);
+            } else {
+                return sc_error!("Invalid function to call");
+            }
+        }
+
+        while !residuum_vec.is_empty() {
+            let residuum = residuum_vec.pop().unwrap();
```

Comment:
`unwrap` generates "panic" code even if this is always valid. Please use `unwrap_or_default` or `unwrap_or_else` instead.

Reply:
How can it generate panic code when there's a check for empty value above?

Reply:
done